plugins/chat/spec/serializer/chat/structured_channel_serializer_spec.rb

```diff
@@ -104,4 +104,103 @@ def fetch_data
       ],
     ).to eq(nil)
   end
+
+  describe "#meta" do
+    context "when user is anonymous" do
+      it "does not query MessageBus for the user_tracking_state_message_bus_channel last_id" do
+        Chat::Publisher.expects(:user_tracking_state_message_bus_channel).never
+        json = described_class.new(fetch_data, scope: Guardian.new).as_json
+        expect(
+          json.dig(:structured_channel, :meta, :message_bus_last_ids).key?(:user_tracking_state),
+        ).to eq(false)
+      end
+    end
+
+    context "when user is not anonymous" do
+      it "has the required message_bus_last_ids" do
+        expect(
+          described_class
+            .new(fetch_data, scope: guardian)
+            .as_json
+            .dig(:structured_channel, :meta, :message_bus_last_ids)
+            .keys,
+        ).to eq(
+          %i[
+            channel_metadata
+            channel_edits
+            channel_status
+            new_channel
+            archive_status
+            user_tracking_state
+          ],
+        )
+      end
+
+      it "calls MessageBus.last_ids with all the required channels for each public and DM chat chat channel" do
+        MessageBus
+          .expects(:last_ids)
+          .with(
+            Chat::Publisher::CHANNEL_METADATA_MESSAGE_BUS_CHANNEL,
+            Chat::Publisher::CHANNEL_EDITS_MESSAGE_BUS_CHANNEL,
+            Chat::Publisher::CHANNEL_STATUS_MESSAGE_BUS_CHANNEL,
+            Chat::Publisher::NEW_CHANNEL_MESSAGE_BUS_CHANNEL,
+            Chat::Publisher::CHANNEL_ARCHIVE_STATUS_MESSAGE_BUS_CHANNEL,
+            Chat::Publisher.user_tracking_state_message_bus_channel(user1.id),
+            Chat::Publisher.new_messages_message_bus_channel(channel1.id),
+            Chat::Publisher.new_mentions_message_bus_channel(channel1.id),
+            Chat::Publisher.kick_users_message_bus_channel(channel1.id),
+            Chat::Publisher.root_message_bus_channel(channel1.id),
+            Chat::Publisher.new_messages_message_bus_channel(channel2.id),
+            Chat::Publisher.new_mentions_message_bus_channel(channel2.id),
+            Chat::Publisher.kick_users_message_bus_channel(channel2.id),
+            Chat::Publisher.root_message_bus_channel(channel2.id),
+            Chat::Publisher.new_messages_message_bus_channel(channel3.id),
+            Chat::Publisher.new_mentions_message_bus_channel(channel3.id),
+            Chat::Publisher.root_message_bus_channel(channel3.id),
+            Chat::Publisher.new_messages_message_bus_channel(channel4.id),
+            Chat::Publisher.new_mentions_message_bus_channel(channel4.id),
+            Chat::Publisher.root_message_bus_channel(channel4.id),
+          )
+          .returns({})
+        described_class.new(fetch_data, scope: guardian).as_json
```

Comment:
I would expect `described_class.new(fetch_data, scope: guardian).as_json` to eq that return value above for completion's sake.

Reply:
Yep sure

Reply:
Oh wait that's not correct though -- the point of this spec is to just check that `MessageBus.last_ids` is being called with all the expected channels, and the other tests are similar so the return value doesn't matter too much.